xmltojsonlegacy/src/main/java/org/assimbly/xmltojsonlegacy/processor/XmlToJsonProcessor.java

```diff
@@ -0,0 +1,359 @@
+                    isRootArray = false;
+                }
+            }
+            if(level == 3 && numberOfSiblings == 1 &&
+                    (parentClass==null || !parentClass.equalsIgnoreCase(JSON_XML_ATTR_CLASS_OBJECT))
+            ) {
+                isRootArray = true;
+                isOneValue = true;
+            }
+        } else {
+            if(level == 0 && numberOfChildren == 1) {
+                isRootArray = true;
+            }
+            if(level == 2 && parentClass!=null && parentClass.equalsIgnoreCase(JSON_XML_ATTR_CLASS_ARRAY)) {
+                isRootArray = true;
+            }
+            if(level == 2 && classAttr!=null && classAttr.equalsIgnoreCase(JSON_XML_ATTR_CLASS_OBJECT)) {
+                isObject = true;
+                if(parentSiblings > 1) {
+                    isRootArray = true;
+                } else {
+                    isRootArray = false;
+                }
+            }
+            if(level == 2 && numberOfChildren == 1
+                    && (classAttr==null || !classAttr.equalsIgnoreCase(JSON_XML_ATTR_CLASS_OBJECT))
+            ) {
+                isSingleChildren = true;
+            }
+            if(level == 3 && numberOfSiblings == 1
+                    && (parentClass==null || !parentClass.equalsIgnoreCase(JSON_XML_ATTR_CLASS_OBJECT))
+            ) {
+                isOneValue = true;
+            }
+            if(level == 3 && parentClass!=null && parentClass.equalsIgnoreCase(JSON_XML_ATTR_CLASS_OBJECT)) {
+                isAttributeObject = true;
+            }
+        }
+
+        printElementDetails(
+                element, level, parentClass, parentSiblings, isRootArray, isOneValue, isObject, isFirstChild,
+                numberOfSiblings, numberOfChildren, classAttr, typeAttr
+        );
+
+        if(!xmlJsonDataFormat.isTypeHints() && !isAttributeObject && element.hasAttributes()){
+            NamedNodeMap attrMap = element.getAttributes();
+            for (int j = 0; j < attrMap.getLength(); j++) {
+                Node node = attrMap.item(j);
+                rootObjectNode.put(JSON_XML_ATTR_PREFIX+node.getNodeName(), node.getNodeValue());
+            }
+        }
+
+        // Check if the current element has child nodes
+        if (element.hasChildNodes()) {
+            NodeList nodeList = element.getChildNodes();
+            int nodeListSize = nodeList.getLength();
+            int nodeCount = 0;
+            for (int i = 0; i < nodeListSize; i++) {
+                Node childNode = nodeList.item(i);
+
+                // Process only element nodes (ignore text nodes, comments, etc.)
+                if (childNode.getNodeType() == Node.ELEMENT_NODE) {
+                    nodeCount++;
+                    boolean isFirstSibling = isFirstSiblingByNumCounts(nodeCount);
+                    Element childElement = (Element) childNode;
+
+                    if(isObject) {
+                        printData(" 1. IS OBJECT", level);
```

Comment:
Note sure what printData does. Is this for debugging. Why not use a logger instead of System.out?  Maybe good to document the code what is does and why it is needed.

Reply:
Yes, it was only for debug purpose. I already changed it.